### ISSUE TITLE
NVDA does not announce li elements in list on alert

### ISSUE THREAD

**Issue by csardo** [nvaccess/nvda#8557]
### Steps to reproduce:
https://jsfiddle.net/91y2odvw/6/

NVDA does not read all of the li elements in a list <ul> element when shown with role="alert". Narrator simply reads "Alert".

If the <li> elements are changed to <div> or <content> tags, then NVDA can read the errors in the list. 

### Actual behavior:
NVDA only announces "Alert" when error is shown

### Expected behavior:
NVDA announces "Alert: Error 1, Error 2" when error is shown

### System configuration:
Microsoft Edge 42.17134.1.0,  FireFox 62.0b11, Chrome Version 67.0.3396.99

#### NVDA version:
2017.4

#### Windows version:
10.0.17134 

#### Name and version of other software in use when reproducing the issue:
Microsoft Edge 42.17134.1.0,  FireFox 62.0b11, Chrome Version 67.0.3396.99

#### Does the issue still occur after restarting your PC?
Yes 

#### Have you tried any other versions of NVDA?
No

**Comment by josephsl** [comment]
Hi,

A few things:

1. Is this specific to Edge? How about with Firefox?
2. Does this happen with NVDA 2018.2.1 also?
3. How about if you install Windows 10 App Essentials? Does the problem happen even with this add-on installed?

Thanks.

**Comment by josephsl** [comment]
Hi,

I was able to reproduce this.

Technical: system alert event is raised by Edge. Unfortunately, this is one of those that has no name but the content is scattered across its children, making this a variant of #8466. Whereas that issue deals with live region changed event, this one has to do with system alert event.

I'll cook up a solution as part of Windows 10 App Essentials 18.08 and sand morph pull request for #8466 accordingly.

Thanks for reporting this. CC @MichaelDCurran

**Comment by josephsl** [comment]
Hi,

Update: yes, fix for #8466 will also take care of this issue as well. I'll update the pull request accordingly.

Thanks.

**Comment by Adriani90** [comment]
Closing due to same reason as stated in #8466.

**Comment by jmpainter** [comment]
@Adriani90 @josephsl @csardo 
role=alert is now announcing in Edge. However, it looks like the issue described here where list items in an alert are not being announced is still present. (using NVDA version 2021.1) I'm not part of this project - my apologies for bringing up an old issue and thanks for all that you do.

**Comment by scottaohara** [comment]
@jmpainter i cannot reproduce with 2021.3 with this test case https://codepen.io/scottohara/pen/wvrLRzY and latest Edge (far different than the Edge this bug was filed against).

if my markup patterns differ from yours, could you provide a reduced test case where you can reproduce the issue?

**Comment by jmpainter** [comment]
The button in the provided codepen, "test 1" with role=alert does not announce the unordered list in Firefox 91.3.0. I can't update to the latest Firefox on my work machine. This may be fixed with the latest Firefox version.

**Comment by scottaohara** [comment]
interesting.  it does with chromium browsers (edge per the original purpose of this ticket) with nvda.  and it does with firefox + voice over.

**Comment by jmpainter** [comment]
Yes, it came because our target browser + screen reader pairs are Firefox/NVDA, Chrome/Jaws, and Safari/VoiceOver.

**Comment by scottaohara** [comment]
gotcha.  so it sounds like you have a similar, but separate issue then, being it's specific to firefox/nvda.

**Comment by jmpainter** [comment]
Yes, that's true because in this issue the system configuration is Microsoft Edge.
